woocommerce/google-listings-and-ads
# Update googleads library to v10

### Changes proposed in this Pull Request:

Update the Google Ads library to v10
Makes use of the Google Ads API v8

```
composer update googleads/google-ads-php --with-all-dependencies

- Upgrading google/gax (1.7.0 => v1.7.1)
- Upgrading google/protobuf (v3.15.6 => v3.17.3)
- Upgrading googleads/google-ads-php (v8.0.0 => v10.0.0)
- Upgrading grpc/grpc (1.36.0 => 1.38.0)
```

Todo: We still need to switch the WCS to using API version v8. This is used for the requests to create an ads account and link an existing ads account.

Closes #588 

### Detailed test instructions:

1. Checkout this PR
2. Run `rm -rf vendor && composer install`
3. Run `ls vendor/googleads/google-ads-php/metadata/Google/Ads/GoogleAds`
4. Confirm we only see V8
5. On the connection test page test, get customers from Google Ads
6. Test list campaigns and create / edit / delete campaign
7. Disconnect ads account
8. Connect existing ads account (tests linking to merchant and creating conversionAction)
9. Test report pages
10. Confirm that we don't see any warnings or errors when sending any of these requests to the Ads API

### Changelog entry
* Update - Version 10.0 of the Google Ads library
